tests/ui/could_be_assoc_type_bounds.fixed

```diff
@@ -0,0 +1,95 @@
+//@aux-build:proc_macros.rs
+#![allow(clippy::extra_unused_type_parameters)]
+
+extern crate proc_macros;
+
+fn projection_with_existing_assoc_bounds<T>()
+where
+    T: Iterator<Item: Clone + Copy + Sized>,
+    
+    //~^ could_be_assoc_type_bounds
+{
+}
+
+fn projection_with_existing_bounds<T: Iterator<Item: Clone + Copy + Sized>>()
+    //~^ could_be_assoc_type_bounds
+{
+}
+
+fn no_fully_qualified_path<T: Iterator<Item: Clone>>()
+where
+    // False negative for now: `T::Item` has a `Res::Err` resolution
+    T::Item: Copy + Sized,
+{
+}
+
+fn ty_param<T: Iterator<Item: Clone>, >() {}
```

Comment:
One case where the lint relies on rustfmt to cleanup. Naively removing the leading or trailing comma would create overlapping spans when trying to remove two bounds next to each other, so that's why it only tries to remove trailing commas